pkg/web/api/connector_v1.go

```diff
@@ -65,6 +68,33 @@ func (c *ConnectorAPIv1) ListConnectors(
 	return &apiv1.ListConnectorsResponse{Connectors: clist}, nil
 }
 
+func (c *ConnectorAPIv1) InspectConnector(req *apiv1.InspectConnectorRequest, server apiv1.ConnectorService_InspectConnectorServer) error {
+	if req.Id == "" {
+		return status.ConnectorError(cerrors.ErrEmptyID)
+	}
+
+	session, err := c.cs.Inspect(server.Context(), req.Id)
+	if err != nil {
+		return status.ConnectorError(cerrors.Errorf("failed to get connector: %w", err))
+	}
+
+	for rec := range session.C {
+		recProto, err2 := toproto.Record(rec)
+		if err2 != nil {
+			return cerrors.Errorf("failed converting record: %w", err2)
+		}
+
+		err2 = server.Send(&apiv1.InspectConnectorResponse{
+			Record: recProto,
+		})
+		if err2 != nil {
+			return cerrors.Errorf("failed sending record: %w", err2)
+		}
+	}
+
+	return cerrors.Errorf("inspector session closed: %w", session.CloseReason())
```

Comment:
Related to the previous comment - it still seems like there's no way to gracefully stop an inspector without getting an error 😕

Reply:
When you said "stop an inspector" I believe what you meant is the session actually? Or you meant the inspector in a wider sense?

An inspector session can be closed either by terminating the WebSocket connection itself (in which case a client won't get any error anyway) or by sending a `close` packet, in which case the client will get an error once the connection has actually been closed.

Reply:
The thing with that is that (at least at the moment) we don't know if the context cancelation was triggered by a `close` packet or something else (maybe an internal thing, since it's bound to the gRPC stream).